.github/workflows/haskell.yml

```diff
@@ -0,0 +1,57 @@
+name: Haskell CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    defaults:
+      run:
+        shell: bash
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ["8.6.5", "8.10.2"]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        exclude:
+          - os: windows-latest
+            ghc: 8.6.5
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Select build directory
+      run: echo "CABAL_BUILDDIR=dist" >> $GITHUB_ENV
+
+    - uses: actions/setup-haskell@v1
+      id: setup-haskell
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: '3.2.0.0'
+
+    - name: Cabal update
+      run: cabal update
+
+    - name: Cabal Configure
+      run: cabal configure --builddir="$CABAL_BUILDDIR" --enable-tests --enable-benchmarks --write-ghc-environment-files=always
+
+    - uses: actions/cache@v2
+      if: matrix.os != 'macos-latest'
```

Comment:
Why not?

Reply:
The cache doesn't work on MacOS.  If I enable the cache, the files are restored in such a way that the executables are missing and tests fail.  I still haven't figured it out.  I'll let you know when I do.

Reply:
Just testing this again now.

Reply:
Looks like the cache is working now too.  Maybe the Github Action that does the caching got fixed.